rest-server/src/models/vc.js

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation
+// All rights reserved.
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+// module dependencies
+const unirest = require('unirest');
+const yarnConfig = require('../config/yarn');
+const logger = require('../config/logger');
+
+class Vc {
+  constructor(name, next) {
+    this.getVcList((vcList, error) => {
+      if (error === null) {
+        if (name in vcList) {
+          for (let key of Object.keys(vcList[name])) {
+            this[key] = vcList[name][key];
+          }
+        } else {
+          error = new Error('VcNotFound');
+        }
+      }
+      next(this, error);
+    });
+  }
+
+  getCapacitySchedulerInfo(queueInfo) {
+    let queues = {};
+    function traverse(queueInfo, queueDict) {
```

Comment:
Could you explain why use a recursive way to iterate all queues?

Reply:
The capacity scheduler supports hierarchical queues.

Reply:
I got it. Good idea. However, we do not have such a complicated design yet, and we won't have such a design in the foreseeable future. So I suggest we make things simpler -- simplicity usually results in higher performance, higher robustness, and higher maintainability.